app/ante/authz_test.go

```diff
@@ -0,0 +1,550 @@
+					stakingAuthDelegate,
+					&distantFuture,
+				),
+			},
+			false,
+			nil,
+		},
+		{
+			"disabled msg - MsgGrant contains a blocked msg",
+			[]sdk.Msg{
+				newMsgGrant(
+					testAddresses[0],
+					testAddresses[1],
+					authz.NewGenericAuthorization(sdk.MsgTypeURL(&evmtypes.MsgEthereumTx{})),
+					&distantFuture,
+				),
+			},
+			false,
+			sdkerrors.ErrUnauthorized,
+		},
+		{
+			"disabled msg - MsgGrant contains a blocked msg",
+			[]sdk.Msg{
+				newMsgGrant(
+					testAddresses[0],
+					testAddresses[1],
+					stakingAuthUndelegate,
+					&distantFuture,
+				),
+			},
+			false,
+			sdkerrors.ErrUnauthorized,
+		},
+		{
+			"allowed msg - when a MsgExec contains a non blocked msg",
+			[]sdk.Msg{
+				newMsgExec(
+					testAddresses[1],
+					[]sdk.Msg{banktypes.NewMsgSend(
+						testAddresses[0],
+						testAddresses[3],
+						sdk.NewCoins(sdk.NewInt64Coin(evmtypes.DefaultEVMDenom, 100e6)),
+					)}),
+			},
+			false,
+			nil,
+		},
+		{
+			"disabled msg - MsgExec contains a blocked msg",
+			[]sdk.Msg{
+				newMsgExec(
+					testAddresses[1],
+					[]sdk.Msg{
+						&evmtypes.MsgEthereumTx{},
+					},
+				),
+			},
+			false,
+			sdkerrors.ErrUnauthorized,
+		},
+		{
+			"disabled msg - surrounded by valid msgs",
+			[]sdk.Msg{
+				newMsgGrant(
+					testAddresses[0],
+					testAddresses[1],
+					stakingAuthDelegate,
+					&distantFuture,
+				),
+				newMsgExec(
+					testAddresses[1],
+					[]sdk.Msg{
+						banktypes.NewMsgSend(
+							testAddresses[0],
+							testAddresses[3],
+							sdk.NewCoins(sdk.NewInt64Coin(evmtypes.DefaultEVMDenom, 100e6)),
+						),
+						&evmtypes.MsgEthereumTx{},
+					},
+				),
+			},
+			false,
+			sdkerrors.ErrUnauthorized,
+		},
+		{
+			"disabled msg - nested MsgExec containing a blocked msg",
+			[]sdk.Msg{
+				createNestedMsgExec(
+					testAddresses[1],
+					2,
+					[]sdk.Msg{
+						&evmtypes.MsgEthereumTx{},
+					},
+				),
+			},
+			false,
+			sdkerrors.ErrUnauthorized,
+		},
+		{
+			"disabled msg - nested MsgGrant containing a blocked msg",
+			[]sdk.Msg{
+				newMsgExec(
+					testAddresses[1],
+					[]sdk.Msg{
+						newMsgGrant(
+							testAddresses[0],
+							testAddresses[1],
+							authz.NewGenericAuthorization(sdk.MsgTypeURL(&evmtypes.MsgEthereumTx{})),
+							&distantFuture,
+						),
+					},
+				),
+			},
+			false,
+			sdkerrors.ErrUnauthorized,
+		},
+		{
+			"disabled msg - nested MsgExec NOT containing a blocked msg but has more nesting levels than the allowed",
+			[]sdk.Msg{
+				createNestedMsgExec(
+					testAddresses[1],
+					6,
+					[]sdk.Msg{
+						banktypes.NewMsgSend(
+							testAddresses[0],
+							testAddresses[3],
+							sdk.NewCoins(sdk.NewInt64Coin(evmtypes.DefaultEVMDenom, 100e6)),
+						),
+					},
+				),
+			},
+			false,
+			sdkerrors.ErrUnauthorized,
+		},
+		{
+			"disabled msg - multiple two nested MsgExec messages NOT containing a blocked msg over the limit",
+			[]sdk.Msg{
+				createNestedMsgExec(
```

Comment:
Code is duplicated, can refactor to a util:

e.g. `createNestedExecMsgSend(depth int)`